src/betterproto/__init__.py

```diff
@@ -1493,24 +1493,30 @@ def to_dict(
                     else:
                         output[cased_name] = b64encode(value).decode("utf8")
                 elif meta.proto_type == TYPE_ENUM:
+                    def name(enum_class, value):
+                        obj = enum_class(value)
+                        if hasattr(obj.__class__, 'full_name') and isinstance(obj.__class__.full_name, property):
+                            return obj.full_name
+                        return obj.name
```

Comment:
```suggestion
                    def name(enum_class, value):
                        obj = enum_class(value)
                        if hasattr(obj.__class__, 'full_name') and isinstance(obj.__class__.full_name, property):
                            return obj.full_name
                        return obj.name
```
This should probably go in utils or enum.py